sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BatchScanExec.scala

```diff
@@ -37,7 +38,8 @@ case class BatchScanExec(
     @transient scan: Scan,
     runtimeFilters: Seq[Expression],
     keyGroupedPartitioning: Option[Seq[Expression]] = None,
-    ordering: Option[Seq[SortOrder]] = None) extends DataSourceV2ScanExecBase {
+    ordering: Option[Seq[SortOrder]] = None,
+    @transient table: Option[Table] = None) extends DataSourceV2ScanExecBase {
```

Comment:
for the test, shall we just pass a fake table? e.g. `new TestLocalScanTable("fake")`. Then this doesn't need to be an option.

Reply:
Sounds good - changed in the latest commit.